components/admin_console/filter/filter_checkbox.tsx

```diff
@@ -11,24 +11,38 @@ type Props = {
 };
 
 class FilterCheckbox extends React.PureComponent<Props> {
-    toggleOption = () => {
+    toggleOption = (e: React.MouseEvent) => {
+        e.preventDefault();
+        e.stopPropagation();
         const {checked, name, updateOption} = this.props;
         updateOption(!checked, name);
     }
 
     render() {
         const {name, checked, label} = this.props;
         return (
-            <div className='FilterList_checkbox'>
+            <div
+                className='FilterList_checkbox'
+                onClick={this.toggleOption}
+            >
                 <label>
-                    <input
-                        type='checkbox'
-                        id={name}
-                        name={name}
-                        checked={checked}
-                        onChange={this.toggleOption}
-                    />
+                    {checked &&
+                        <input
+                            type='checkbox'
+                            id={name}
+                            name={name}
+                            defaultChecked={true}
+                        />
```

Comment:
This might seem weird but I was having issues where the input checkbox would not get updated in the DOM when rendered inside of the bootstrap modal... my assumption is that somewhere something was calling a `stopPropogation` or `preventDefault` on the click event, so instead of relying on the browser checkbox behaviour i've modified this to simply swap out the input box to a checked one when it is checked and the opposite when not checked

Reply:
yea... this.props.checked is being updated correctly but the actual checkbox itself does not get re-rendered, im not sure whats going on but this implementation if simply having two different checkboxes for a checked and unchecked state works fine... unless you got any suggestions as to why the checkbox wasnt updating before.